### ISSUE TITLE
HwndHost fail to end thread model state

### ISSUE THREAD

**Issue by tbscer** [dotnet/wpf#8172]
### Description

I have a WPF app, it needs to embed another app which is written as a Winform app. So I used HwndHost to embed it.
It works fine. I used .Net Framework 4.7.2

But when the Winform App shows a model form, such as a message box, and at that moment, i kill the process of Winform App in system TaskManager, without clicking the message box button.  Then the WPF app will not reponse to any mouse click nor keyboard input.

I think the reason is the winform app stop at a model state, and the WPF app can't quit that state.

### Reproduction Steps

1. write a WPF app, and a HwndHost control
2. write a Winform app, it has a button to show message box
3. embed the Winform app. click the button to show message box, without cliking any confirm/cancel button
4. kill the process of winform app in TaskManager
5. then the wpf app hangs

### Expected behavior

when the winform app ends, the wpf app should run normally to response any mouse or keyboard input

### Actual behavior

the wpf app will hang, i have to kill the process of WPF app

### Known Workarounds

I tried to exit thread model state, but I don't know how to do it

### Configuration

Windows 10, .net framework 4.7.2

**Comment by singhashish-wpf** [comment]
@tbscer can you please attach a sample repro for us to investigate?

**Comment by miloush** [comment]
I wouldn't think we are guaranteeing or want to gauranteee an uncorrupted state when users forcibly terminate processes (which can happen at random point). If it does not repro when the winform app closes normally, it isn't worth spending time on.

**Comment by tbscer** [comment]
> yes, I just upload a very simple test code for you.

https://github.com/tbscer/HwndHostTest

**Comment by tbscer** [comment]
> 

i am not sure. I think the HwndHost should provide to solution to end that state.

**Comment by tbscer** [comment]
> @tbscer can you please attach a sample repro for us to investigate?

can you pls check the code updated?

**Comment by tbscer** [comment]
@singhashish-wpf can you pls check the code updated?

**Comment by singhashish-wpf** [comment]
@tbscer sure. Will look at it. Could you also share the scenario because you are killing the winforms process. This doesn't appear a valid user scenario (Unable to think of such a case). Would be great if you could also confirm if the issue reproduces when the winform app closes normally as @miloush asked in previous comment.

**Comment by tbscer** [comment]
> @tbscer sure. Will look at it. Could you also share the scenario because you are killing the winforms process. This doesn't appear a valid user scenario (Unable to think of such a case). Would be great if you could also confirm if the issue reproduces when the winform app closes normally as @miloush asked in previous comment.

yes, if the winform app closes normally, no any model form is forced to quit, the WPF HwndHost will not have any problem.  It only occurs when some unhandled exception happned in the winform app, or the winforn app is killed in process manager, when any model form shows.

**Comment by tbscer** [comment]
Yes, we'd test it. If the winform app was closed normally, the Hwndhost will work properly. This issue happend only if the winform app shows a model form, at this time either any unhandle exception occurs caused the winform app crashed, or the winform app process is killed.
